nav2_docking/opennav_docking/include/opennav_docking/simple_charging_dock.hpp

```diff
@@ -103,6 +105,16 @@ class SimpleChargingDock : public opennav_docking_core::ChargingDock
    */
   virtual bool hasStoppedCharging();
 
+  /**
+   * @brief Start external detection process (service call + subscribe).
+   */
+  void startDetectionProcess() override {startDetection();}
```

Comment:
Just implement startDetectionProcess with startDetection's content, there's not a need for a new method.

Ditto with the Stop functions

Reply:
@bkoensgen this was not completed :-)

Reply:
@bkoensgen this was still not completed